src/System.Runtime.Caching/src/System/Runtime/Caching/PhysicalMemoryMonitor.cs

```diff
@@ -78,15 +78,22 @@ internal PhysicalMemoryMonitor(int physicalMemoryLimitPercentage)
 
         protected override int GetCurrentPressure()
         {
-            Interop.Kernel32.MEMORYSTATUSEX memoryStatusEx = default;
-            memoryStatusEx.dwLength = (uint)Marshal.SizeOf(typeof(Interop.Kernel32.MEMORYSTATUSEX));
-            if (Interop.Kernel32.GlobalMemoryStatusEx(out memoryStatusEx) == 0)
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                Interop.Kernel32.MEMORYSTATUSEX memoryStatusEx = default;
+                memoryStatusEx.dwLength = (uint)Marshal.SizeOf(typeof(Interop.Kernel32.MEMORYSTATUSEX));
```

Comment:
It is better to use generic variant of this `Marshal.SizeOf<Interop.Kernel32.MEMORYSTATUSEX>()`. It is more friendly to AOT compilation.

Reply:
Thanks Jan!